packages/@aws-cdk/aws-apprunner-alpha/lib/observability-configuration.ts

```diff
@@ -0,0 +1,169 @@
+import * as cdk from 'aws-cdk-lib/core';
+import { Construct } from 'constructs';
+import { CfnObservabilityConfiguration } from 'aws-cdk-lib/aws-apprunner';
+
+/**
+ * The implementation provider chosen for tracing App Runner services
+ */
+export enum Vendor {
+  /**
+   * AWS X-RAY
+   */
+  AWSXRAY = 'AWSXRAY',
+}
+
+/**
+ * Properties of the AppRunner Observability configuration
+ */
+export interface ObservabilityConfigurationProps {
+  /**
+   * The name for the ObservabilityConfiguration.
+   *
+   * @default - a name generated by CloudFormation
+   */
+  readonly observabilityConfigurationName?: string;
+
+  /**
+   * The implementation provider chosen for tracing App Runner services.
+   *
+   * You can not attach ObservabilityConfiguration with no vendor to the App Runner Service.
+   *
+   * @default - tracing is not enabled for the service
+   */
+  readonly vendor?: Vendor;
```

Comment:
```suggestion
  /**
   * The implementation provider chosen for tracing App Runner services.
   */
  readonly traceConfigurationVendor: TraceConfigurationVendor;
```
Given that the observability configuration needs to be attached to an AppRunner service and https://github.com/aws/aws-cdk/pull/30359#discussion_r1623313858 (thanks for the follow-up), I'd rather make this required to provide a clearer implementation. 
Also, I'd give it a more meaningful name.
Let me know if you think that there are scenarios where we might want this to be `undefined`.

Reply:
Thank you.
I cannot think of any cases where I would want the property to be undefined, so I agree that it should be a required field.